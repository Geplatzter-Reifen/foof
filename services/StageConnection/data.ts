import { MapPoint } from "@/utils/locationUtil";

export const flensburg: MapPoint = {
  latitude: 54.789356,
  longitude: 9.434199,
};
export const oberstdorf: MapPoint = {
  latitude: 47.408333325,
  longitude: 10.27931564,
};
export const hannover: MapPoint = {
  latitude: 52.3744779,
  longitude: 9.7385532,
};
export const wuerzburg: MapPoint = {
  latitude: 49.7933723,
  longitude: 9.9309779,
};
export const wiesbaden: MapPoint = {
  latitude: 52.520007,
  longitude: 13.404954,
};
