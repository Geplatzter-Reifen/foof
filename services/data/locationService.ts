import { database } from "@/database";
import { Q } from "@nozbe/watermelondb";
import { Location, Stage } from "@/database/model/model";

// CREATE

export const createLocation = async (
  stageId: string,
  latitude: number,
  longitude: number,
): Promise<Location> => {
  return database.write(async () => {
    const stage = await database.get<Stage>("stages").find(stageId);
    return database.get<Location>("locations").create((location) => {
      location.stage.set(stage);
      location.latitude = latitude;
      location.longitude = longitude;
      location.recordedAt = Date.now();
    });
  });
};

export const createLocationWithRecordedAt = async (
  stageId: string,
  latitude: number,
  longitude: number,
  recordedAt: number,
): Promise<Location> => {
  return database.write(async () => {
    const stage = await database.get<Stage>("stages").find(stageId);
    return database.get<Location>("locations").create((location) => {
      location.stage.set(stage);
      location.latitude = latitude;
      location.longitude = longitude;
      location.recordedAt = recordedAt;
    });
  });
};

// READ

export const getAllLocationsByStageIdQuery = (stageId: string) => {
  return database
    .get<Location>("locations")
    .query(Q.where("stage_id", stageId));
};
export const getAllLocationsByStageId = (stageId: string) => {
  return getAllLocationsByStageIdQuery(stageId).fetch();
};

//gets the location of a stage the has the earliest "startedAt" timestamp
export const getFirstLocationByTourId = async (tourId: string) => {
  const stages = await database
    .get<Stage>("stages")
    .query(Q.where("tour_id", tourId))
    .fetch();
  const stage = stages.reduce((prev, current) =>
    prev.startedAt < current.startedAt ? prev : current,
  );
  return getAllLocationsByStageId(stage.id);
};
