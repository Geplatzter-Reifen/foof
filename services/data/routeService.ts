import { database } from "@/database";
import { Q } from "@nozbe/watermelondb";
import { Tour, Route } from "@/database/model/model";

// READ

/**
 * @param tourId The id of the tour
 * @returns The route for the tour, or null if no route exists
 */
export const getTourRoute = async (tourId: string) => {
  const route = await database
    .get<Route>("routes")
    .query(Q.where("tour_id", tourId))
    .fetch();
  if (route.length === 0) {
    return null;
  }
  return route[0];
};

// UPDATE

/**
 * Sets the route for a tour. Creates a new route if none exists, updates the existing route otherwise.
 * @param tourId The id of the tour
 * @param geoJson The GeoJSON string of the route
 * @returns The created or updated route
 */
export const setTourRoute = async (tourId: string, geoJson: string) => {
  return database.write(async () => {
    // Check if a route for the tour already exists
    const existingRoute = await database
      .get<Route>("routes")
      .query(Q.where("tour_id", tourId))
      .fetch();

    if (existingRoute.length === 0) {
      // Create a new route
      const tour = await database.get<Tour>("tours").find(tourId);
      return database.get<Route>("routes").create((route) => {
        route.tour.set(tour);
        route.geoJson = geoJson;
      });
    } else {
      // Update the existing route
      return existingRoute[0].update((route) => {
        route.geoJson = geoJson;
      });
    }
  });
};

// DELETE

/** Deletes a Tour's Route
 * @param tourId The id of the tour
 */
export const deleteRoute = async (tourId: string) => {
  void database.write(async () => {
    const routes = await database
      .get<Route>("routes")
      .query(Q.where("tour_id", tourId))
      .fetch();
    routes.forEach((r) => r.destroyPermanently());
  });
};
