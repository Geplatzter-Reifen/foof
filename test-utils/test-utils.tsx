// https://testing-library.com/docs/react-testing-library/setup/#add-custom-queries
import React from "react";
import { render, RenderOptions } from "@testing-library/react-native";
import { FontAwesomeIconsPack } from "@/components/Font/fontAwesome";
import { mapping } from "@eva-design/eva";
import {
  IconRegistry,
  ApplicationProvider,
  ModalService,
} from "@ui-kitten/components";
import * as eva from "@eva-design/eva";
import { foofDarkTheme, foofLightTheme } from "@/constants/custom-theme";
import { library } from "@fortawesome/fontawesome-svg-core";
import { far } from "@fortawesome/free-regular-svg-icons";
import { fas } from "@fortawesome/free-solid-svg-icons";
import { fab } from "@fortawesome/free-brands-svg-icons";

// Iconpacks laden
library.add(far, fas, fab);

// ModalService wird verwendet, um die Modal-Positionierung zu steuern
ModalService.setShouldUseTopInsets = true;

type Props = {
  children: React.ReactNode;
  use_dark_theme?: boolean;
};
// Der Wrapper wird verwendet, um die Provider zu umschließen, die für die Komponenten notwendig sind
const AllTheProviders = ({ children, use_dark_theme = true }: Props) => {
  const theme = use_dark_theme
    ? { ...eva.dark, ...foofDarkTheme }
    : { ...eva.light, ...foofLightTheme };
  return (
    <>
      <IconRegistry icons={FontAwesomeIconsPack} />
      <ApplicationProvider {...eva} customMapping={mapping} theme={theme}>
        {children}
      </ApplicationProvider>
    </>
  );
};

// Erstellt einen Custom-Render, der die Provider um die Komponenten wickelt
const customRender = (
  component: React.ReactElement,
  options?: Omit<RenderOptions, "wrapper">,
) => {
  return render(component, { wrapper: AllTheProviders, ...options });
};

// Exportiert alle Funktionen der Testing Library, weswegen man diese dann auch von dieser Datei importieren kann
export * from "@testing-library/react-native";
export { customRender as render };
