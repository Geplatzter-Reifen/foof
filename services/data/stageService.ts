import { database } from "@/database";
import { Q } from "@nozbe/watermelondb";
import { Stage, Tour } from "@/database/model/model";
import { getAllLocationsByStageId } from "@/services/data/locationService";

// CREATE

export const createStage = async (
  tourId: string,
  title: string,
  startedAt?: number,
  finishedAt?: number,
  active: boolean = false,
  distance?: number,
): Promise<Stage> => {
  // if stage to be active, deactivate all active stages
  if (active) {
    await setActiveStageInactive();
  }

  const tour = await database.get<Tour>("tours").find(tourId);

  // Update Tour Start if necessary
  if (!tour.startedAt) {
    await database.write(async () => {
      await tour.update(() => {
        tour.startedAt = startedAt ?? Date.now();
      });
    });
  } else if (startedAt && startedAt < tour.startedAt) {
    await database.write(async () => {
      await tour.update(() => {
        tour.startedAt = startedAt;
      });
    });
  }

  // Create Stage
  return tour.addStage(title, startedAt, finishedAt, distance, active);
};

/** creates a Stage and sets it active */
export const startStage = async (tourId: string) => {
  const title: string = `Etappe ${(await getAllStagesByTourId(tourId)).length + 1}`;

  return createStage(tourId, title, Date.now(), undefined, true);
};

// READ

export const getAllStagesByTourIdQuery = (tourId: string) => {
  return database.get<Stage>("stages").query(Q.where("tour_id", tourId));
};
export const getAllStagesByTourId = (tourId: string) => {
  return getAllStagesByTourIdQuery(tourId).fetch();
};

export const getStageByStageId = (stageId: string) => {
  return database.get<Stage>("stages").find(stageId);
};

export const getActiveStage = async (): Promise<Stage | null> => {
  const activeStages = await database
    .get<Stage>("stages")
    .query(Q.where("is_active", true), Q.take(1))
    .fetch();
  if (activeStages.length === 0) {
    return null;
  } else {
    return activeStages[0];
  }
};

// UPDATE

export const finishStage = async (stageId: string, finishTime?: number) => {
  void database.write(async () => {
    const stageToFinish = await getStageByStageId(stageId);
    await stageToFinish.update(() => {
      stageToFinish.finishedAt = finishTime ?? Date.now();
      stageToFinish.isActive = false;
    });
  });
};

export const setStageDistance = async (stageId: string, distance: number) => {
  void database.write(async () => {
    const stage = await getStageByStageId(stageId);
    await stage.update(() => {
      stage.distance = distance;
    });
  });
};

export const setStageAvgSpeed = async (stageId: string, speed: number) => {
  void database.write(async () => {
    const stage = await getStageByStageId(stageId);
    await stage.update(() => {
      stage.avgSpeed = speed;
    });
  });
};

const setActiveStageInactive = async () => {
  await database.write(async () => {
    const allStages = await database
      .get<Stage>("stages")
      .query(Q.where("is_active", true))
      .fetch();
    for (const stage of allStages) {
      await stage.update(() => {
        stage.isActive = false;
      });
    }
  });
};

// DELETE

export const deleteStage = async (stageId: string) => {
  void database.write(async () => {
    const stage = await getStageByStageId(stageId);
    const locations = await getAllLocationsByStageId(stage.id);
    for (const location of locations) {
      await location.destroyPermanently();
    }
    await stage.destroyPermanently();
    await stage.destroyPermanently();
  });
};
