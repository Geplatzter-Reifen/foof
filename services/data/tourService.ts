import { Q } from "@nozbe/watermelondb";
import { database } from "@/model/createDatabase";
import { Tour } from "@/model/model";
import { getAllStagesByTourId } from "@/services/data/stageService";
import { getAllLocationsByStageId } from "@/services/data/locationService";

// READ

export const getAllToursQuery = database.get<Tour>("tours").query();
export const getAllTours = () => getAllToursQuery.fetch();

export const getTourByTourIdQuery = (tourId: string) => {
  return database.get<Tour>("tours").query(Q.where("id", tourId));
};
export const getTourByTourId = (tourId: string) => {
  return database.get<Tour>("tours").find(tourId);
};

export const getActiveTour = async (): Promise<Tour | null> => {
  const activeTours = await database
    .get<Tour>("tours")
    .query(Q.where("is_active", true), Q.take(1))
    .fetch();
  if (activeTours.length === 0) {
    return null;
  } else {
    return activeTours[0];
  }
};

// CREATE

export const createTour = async (
  title: string,
  startedAt?: number,
): Promise<Tour> => {
  return database.write(async () => {
    return database.get<Tour>("tours").create((tour) => {
      tour.title = title;
      tour.isActive = false;
      if (startedAt) {
        tour.startedAt = startedAt;
      }
    });
  });
};

// UPDATE

export const setTourActive = async (tourId: string) => {
  await database.write(async () => {
    const allTours = await database.get<Tour>("tours").query().fetch();
    for (const tour of allTours) {
      if (tour.isActive) {
        await tour.update(() => {
          tour.isActive = false;
        });
      }
    }
    const tour = await database.get<Tour>("tours").find(tourId);
    await tour.update(() => {
      tour.isActive = true;
    });
  });
};

export const setTourInactive = async (tourId: string) => {
  await database.write(async () => {
    const tour = await database.get<Tour>("tours").find(tourId);
    await tour.update(() => {
      tour.isActive = false;
    });
  });
};

export const updateTourNameById = async (
  tourId: string,
  newTourName: string,
) => {
  await database.write(async () => {
    const tour = await getTourByTourId(tourId);
    await tour.update(() => {
      tour.title = newTourName;
    });
  });
};

// DELETE

export const deleteTour = (tourId: string) => {
  void database.write(async () => {
    const tour = await getTourByTourId(tourId);
    const stages = await getAllStagesByTourId(tour.id);
    for (const stage of stages) {
      const locations = await getAllLocationsByStageId(stage.id);
      for (const location of locations) {
        await location.destroyPermanently();
      }
      await stage.destroyPermanently();
    }
    await tour.destroyPermanently();
  });
};

export const deleteAllTours = () => {
  void database.write(async () => {
    const tours = await getAllTours();
    for (const tour of tours) {
      const stages = await getAllStagesByTourId(tour.id);
      for (const stage of stages) {
        const locations = await getAllLocationsByStageId(stage.id);
        for (const location of locations) {
          await location.destroyPermanently();
        }
        await stage.destroyPermanently();
      }
      await tour.destroyPermanently();
    }
  });
};
