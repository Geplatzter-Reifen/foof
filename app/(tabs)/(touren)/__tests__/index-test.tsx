import Touruebersicht from "..";
import { render } from "@/test-utils/test-utils";
import { createTour, setTourActive } from "@/services/data/tourService";
import { waitFor } from "@testing-library/react-native";

describe("Touruebersicht", () => {
  it("should render correctly", async () => {
    const tour = await createTour("Coole Tour");
    await setTourActive(tour.id);
    const view = await waitFor(() => render(<Touruebersicht />));

    expect(view).toMatchSnapshot();
  });
  // Mehr Tests braucht es nicht, da dafÃ¼r bereits Tests der Unterkomponenten existieren (TourStats, StageList)
  // Deshalb hier auch keine besonderen Testdaten
});
