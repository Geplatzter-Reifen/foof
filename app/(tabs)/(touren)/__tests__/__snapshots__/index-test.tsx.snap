// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Touruebersicht should render correctly 1`] = `
<View
  appearance="default"
  level="2"
  style={
    [
      {
        "backgroundColor": "#eaeaea",
      },
      {
        "flex": 1,
      },
    ]
  }
>
  <View
    appearance="default"
    level="1"
    style={
      [
        {
          "backgroundColor": "#FFFFFF",
        },
        undefined,
      ]
    }
  >
    <View
      appearance="default"
      style={
        [
          {
            "backgroundColor": "#FFFFFF",
            "minHeight": 56,
            "paddingHorizontal": 8,
            "paddingVertical": 8,
          },
          {
            "alignItems": "center",
            "flexDirection": "row",
          },
          {
            "justifyContent": "space-between",
          },
          {
            "marginTop": 0,
          },
        ]
      }
    >
      <View
        style={
          {
            "flexDirection": "row",
            "zIndex": 1,
          }
        }
      >
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          hitSlop={15}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "marginHorizontal": 8,
              "opacity": 1,
            }
          }
        >
          <View
            collapsable={false}
            style={
              {
                "transform": [
                  {
                    "scale": 1,
                  },
                ],
              }
            }
          >
            <Text
              allowFontScaling={false}
              selectable={false}
              style={
                [
                  {
                    "color": "#000000",
                    "fontSize": 24,
                  },
                  {
                    "width": "auto",
                  },
                  {
                    "fontFamily": "FontAwesome6Free-Regular",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  {
                    "fontWeight": "400",
                  },
                ]
              }
            >
              
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          {
            "alignItems": "center",
            "bottom": 0,
            "justifyContent": "center",
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      >
        <Text
          appearance="default"
          category="h4"
          status="basic"
          style={
            [
              {
                "color": "#000000",
                "fontFamily": "System",
                "fontSize": 26,
                "fontWeight": "800",
              },
              undefined,
            ]
          }
        >
          Coole Tour
        </Text>
      </View>
      <View
        style={
          [
            {
              "flexDirection": "row",
              "zIndex": 1,
            },
            undefined,
          ]
        }
      >
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          hitSlop={15}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "marginHorizontal": 8,
              "opacity": 1,
            }
          }
        >
          <View
            collapsable={false}
            style={
              {
                "transform": [
                  {
                    "scale": 1,
                  },
                ],
              }
            }
          >
            <Text
              allowFontScaling={false}
              selectable={false}
              style={
                [
                  {
                    "color": "#000000",
                    "fontSize": 24,
                  },
                  {
                    "width": 24,
                  },
                  {
                    "fontFamily": "FontAwesome6Free-Solid",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  {
                    "fontWeight": "900",
                  },
                ]
              }
            >
              
            </Text>
          </View>
        </View>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          hitSlop={15}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "marginHorizontal": 8,
              "opacity": 1,
            }
          }
        >
          <View
            collapsable={false}
            style={
              {
                "transform": [
                  {
                    "scale": 1,
                  },
                ],
              }
            }
          >
            <Text
              allowFontScaling={false}
              selectable={false}
              style={
                [
                  {
                    "color": "#000000",
                    "fontSize": 24,
                  },
                  {
                    "width": 24,
                  },
                  {
                    "fontFamily": "FontAwesome6Free-Solid",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  {
                    "fontWeight": "900",
                  },
                ]
              }
            >
              
            </Text>
          </View>
        </View>
      </View>
    </View>
    <View
      appearance="default"
      style={
        [
          {
            "backgroundColor": "#b8b8b8",
            "height": 1,
          },
          undefined,
        ]
      }
    />
  </View>
  <View
    style={
      {
        "backgroundColor": "rgba(255, 110, 20, 0.26)",
      }
    }
  >
    <View
      style={
        {
          "margin": 10,
          "marginBottom": 0,
          "padding": 8,
        }
      }
    >
      <View
        appearance="default"
        onLayout={[Function]}
        size="small"
        status="primary"
        style={
          [
            {
              "backgroundColor": "rgba(51, 102, 255, 0.24)",
              "borderRadius": 6,
              "height": 27,
            },
            {
              "overflow": "hidden",
            },
            {
              "backgroundColor": "#eaeaea",
              "borderRadius": 6,
              "height": 27,
            },
          ]
        }
      >
        <View
          collapsable={false}
          style={
            {
              "backgroundColor": "#FF6E14",
              "borderRadius": 6,
              "height": 27,
              "width": "0%",
            }
          }
        />
      </View>
      <Text
        appearance="default"
        category="p1"
        status="basic"
        style={
          [
            {
              "color": "#000000",
              "fontFamily": "System",
              "fontSize": 15,
              "fontWeight": "400",
            },
            {
              "color": "#000000",
              "fontSize": 19,
              "left": "4.00%",
              "position": "absolute",
              "top": "28%",
            },
          ]
        }
      >
        0.0%
      </Text>
    </View>
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "height": "auto",
          "paddingBottom": 10,
          "paddingHorizontal": 20,
        }
      }
    >
      <View
        style={
          {
            "alignItems": "flex-start",
            "flex": 1,
            "justifyContent": "flex-start",
          }
        }
      >
        <View
          style={
            {
              "alignItems": "center",
              "flexDirection": "row",
            }
          }
        >
          <View
            collapsable={false}
            style={
              {
                "transform": [
                  {
                    "scale": 1,
                  },
                ],
              }
            }
          >
            <Text
              allowFontScaling={false}
              selectable={false}
              style={
                [
                  {
                    "color": undefined,
                    "fontSize": 15,
                  },
                  {
                    "alignSelf": "center",
                    "marginHorizontal": 4,
                    "minWidth": 17,
                    "width": "100%",
                  },
                  {
                    "fontFamily": "FontAwesome6Free-Solid",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  {
                    "fontWeight": "900",
                  },
                ]
              }
            >
              
            </Text>
          </View>
          <Text
            appearance="default"
            category="p1"
            status="basic"
            style={
              [
                {
                  "color": "#000000",
                  "fontFamily": "System",
                  "fontSize": 15,
                  "fontWeight": "400",
                },
                undefined,
              ]
            }
          >
            0.0 km
          </Text>
        </View>
        <View
          style={
            {
              "alignItems": "center",
              "flexDirection": "row",
            }
          }
        >
          <View
            collapsable={false}
            style={
              {
                "transform": [
                  {
                    "scale": 1,
                  },
                ],
              }
            }
          >
            <Text
              allowFontScaling={false}
              selectable={false}
              style={
                [
                  {
                    "color": undefined,
                    "fontSize": 15,
                  },
                  {
                    "alignSelf": "center",
                    "marginHorizontal": 4,
                    "minWidth": 17,
                    "width": "100%",
                  },
                  {
                    "fontFamily": "FontAwesome6Free-Solid",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  {
                    "fontWeight": "900",
                  },
                ]
              }
            >
              
            </Text>
          </View>
          <Text
            appearance="default"
            category="p1"
            status="basic"
            style={
              [
                {
                  "color": "#000000",
                  "fontFamily": "System",
                  "fontSize": 15,
                  "fontWeight": "400",
                },
                undefined,
              ]
            }
          >
            0 m
          </Text>
        </View>
        <View
          style={
            {
              "alignItems": "center",
              "flexDirection": "row",
            }
          }
        >
          <View
            collapsable={false}
            style={
              {
                "transform": [
                  {
                    "scale": 1,
                  },
                ],
              }
            }
          >
            <Text
              allowFontScaling={false}
              selectable={false}
              style={
                [
                  {
                    "color": undefined,
                    "fontSize": 15,
                  },
                  {
                    "alignSelf": "center",
                    "marginHorizontal": 4,
                    "minWidth": 17,
                    "width": "100%",
                  },
                  {
                    "fontFamily": "FontAwesome6Free-Solid",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  {
                    "fontWeight": "900",
                  },
                ]
              }
            >
              
            </Text>
          </View>
          <Text
            appearance="default"
            category="p1"
            status="basic"
            style={
              [
                {
                  "color": "#000000",
                  "fontFamily": "System",
                  "fontSize": 15,
                  "fontWeight": "400",
                },
                undefined,
              ]
            }
          >
            0.0 km/h
          </Text>
        </View>
        <View
          style={
            {
              "alignItems": "center",
              "flexDirection": "row",
            }
          }
        >
          <View
            collapsable={false}
            style={
              {
                "transform": [
                  {
                    "scale": 1,
                  },
                ],
              }
            }
          >
            <Text
              allowFontScaling={false}
              selectable={false}
              style={
                [
                  {
                    "color": undefined,
                    "fontSize": 15,
                  },
                  {
                    "alignSelf": "center",
                    "marginHorizontal": 4,
                    "minWidth": 17,
                    "width": "100%",
                  },
                  {
                    "fontFamily": "FontAwesome6Free-Solid",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  {
                    "fontWeight": "900",
                  },
                ]
              }
            >
              
            </Text>
          </View>
          <Text
            appearance="default"
            category="p1"
            status="basic"
            style={
              [
                {
                  "color": "#000000",
                  "fontFamily": "System",
                  "fontSize": 15,
                  "fontWeight": "400",
                },
                undefined,
              ]
            }
          >
            0 kcal
          </Text>
        </View>
      </View>
      <View
        style={
          {
            "alignItems": "flex-start",
            "flex": 1,
            "justifyContent": "flex-start",
          }
        }
      >
        <View
          style={
            {
              "alignItems": "center",
              "flexDirection": "row",
            }
          }
        >
          <View
            collapsable={false}
            style={
              {
                "transform": [
                  {
                    "scale": 1,
                  },
                ],
              }
            }
          >
            <Text
              allowFontScaling={false}
              selectable={false}
              style={
                [
                  {
                    "color": undefined,
                    "fontSize": 15,
                  },
                  {
                    "alignSelf": "center",
                    "marginHorizontal": 4,
                    "minWidth": 17,
                    "width": "100%",
                  },
                  {
                    "fontFamily": "FontAwesome6Free-Regular",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  {
                    "fontWeight": "400",
                  },
                ]
              }
            >
              
            </Text>
          </View>
          <Text
            appearance="default"
            category="p1"
            status="basic"
            style={
              [
                {
                  "color": "#000000",
                  "fontFamily": "System",
                  "fontSize": 15,
                  "fontWeight": "400",
                },
                undefined,
              ]
            }
          >
            --
          </Text>
        </View>
        <View
          style={
            {
              "alignItems": "center",
              "flexDirection": "row",
            }
          }
        >
          <View
            collapsable={false}
            style={
              {
                "transform": [
                  {
                    "scale": 1,
                  },
                ],
              }
            }
          >
            <Text
              allowFontScaling={false}
              selectable={false}
              style={
                [
                  {
                    "color": undefined,
                    "fontSize": 15,
                  },
                  {
                    "alignSelf": "center",
                    "marginHorizontal": 4,
                    "minWidth": 17,
                    "width": "100%",
                  },
                  {
                    "fontFamily": "FontAwesome6Free-Regular",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  {
                    "fontWeight": "400",
                  },
                ]
              }
            >
              
            </Text>
          </View>
          <Text
            appearance="default"
            category="p1"
            status="basic"
            style={
              [
                {
                  "color": "#000000",
                  "fontFamily": "System",
                  "fontSize": 15,
                  "fontWeight": "400",
                },
                undefined,
              ]
            }
          >
            0h 0m
          </Text>
        </View>
      </View>
    </View>
  </View>
  <View
    onLayout={[Function]}
    style={
      [
        {
          "flex": 1,
          "flexDirection": "column",
        },
        undefined,
        {
          "flexDirection": "column",
        },
      ]
    }
  >
    <RCTScrollView
      allowDivider={false}
      allowEndFade={true}
      allowStartFade={true}
      fadeColors={
        [
          "rgba(234, 234, 234, 0.3)",
          "rgba(234, 234, 234, 0.9)",
        ]
      }
      fadeSize={15}
      horizontal={false}
      innerRef={null}
      isRtl={false}
      onContentSizeChange={[Function]}
      onScroll={[Function]}
      scrollEventThrottle={16}
      scrollThreshold={10}
      style={
        [
          {
            "flex": 1,
          },
          undefined,
        ]
      }
    >
      <View>
        <View
          style={
            {
              "marginTop": 15,
            }
          }
        >
          <Text
            appearance="default"
            category="p1"
            status="basic"
            style={
              [
                {
                  "color": "#000000",
                  "fontFamily": "System",
                  "fontSize": 15,
                  "fontWeight": "400",
                },
                {
                  "fontSize": 20,
                  "marginTop": 15,
                  "textAlign": "center",
                },
              ]
            }
          >
            Starte eine Etappe!
          </Text>
        </View>
      </View>
    </RCTScrollView>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#FF6E14",
        "borderColor": "#FF6E14",
        "borderRadius": 50,
        "borderWidth": 1,
        "bottom": 35,
        "elevation": 3,
        "flexDirection": "row",
        "height": 80,
        "justifyContent": "center",
        "minHeight": 40,
        "minWidth": 40,
        "opacity": 1,
        "paddingHorizontal": 10,
        "paddingVertical": 12,
        "position": "absolute",
        "right": 15,
        "width": 80,
      }
    }
  >
    <View
      collapsable={false}
      style={
        {
          "transform": [
            {
              "scale": 1,
            },
          ],
        }
      }
    >
      <Text
        allowFontScaling={false}
        selectable={false}
        style={
          [
            {
              "color": "#FFFFFF",
              "fontSize": 40,
            },
            {
              "marginHorizontal": 10,
              "width": "auto",
            },
            {
              "fontFamily": "FontAwesome6Free-Solid",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            {
              "fontWeight": "900",
            },
          ]
        }
      >
        +
      </Text>
    </View>
  </View>
</View>
`;
